### DISCUSSION TITLE
Is there a way to specify a format for the createdField date and the updatedField date?

### DISCUSSION THREAD

**Discussion by ClaytonAndTheClayTones** [sensedeep/dynamodb-onetable#382] [Q&A] [answered]
setting my Schema params as follows:

```
{
    isoDates: true,
    timestamps: true,
    createdField: 'createdAt',
    updatedField: 'updatedAt',
}    
```

results in the following values in my dynamo table:

```
Wed Aug 03 2022 22:34:48 GMT+0000 (Coordinated Universal Time)
```

This isn't the ISO format defined by most libraries and it's not easily searchable or sortable.  My expectation if I set isoDates to true is this ISO compliant format, which is both searchable and sortable and is also what is returned by javascript and several other runtimes natively when rendered as ISO format:

```
2022-08-03T22:34:48.000Z 
```

Is there the ability to use these auto create/updated fields in this format?

**Comment by mobsense** [comment]
Are you supplying the createdField value as a string or as a Date?

This works for us here.

**Reply by ClaytonAndTheClayTones** [reply]
I don't supply a value at all since it's supposed to auto-populate those fields, but I do have the pseudo-column in the onetable Schema set to "String", not "Date" if that's what you mean, but I made that decision because "Date" isn't a primitive type supported by Dynamo iirc. 

```
const Schema = {
  version: '0.0.1',
  indexes: {
    primary: { hash: 'pk', sort: 'sk' }, 
  },
  models: {
    webhookConfiguration: {
      pk: { type: String, value: '${_type}#${id}' },
      sk: { type: String, value: '${_type}' }, 
      id: {
        type: String,
        generate: true,
      },
      clientId: { type: String, required: true },
      createdAt: { type: String },
      updatedAt: { type: String },
    },
  },
  params: {
    isoDates: true,
    timestamps: true,
    createdField: 'createdAt',
    updatedField: 'updatedAt',
  },
}
```

I will try setting that type to Date just in case that _is_ what you're suggesting and I'll report back in if that solves the issue

**Reply by ClaytonAndTheClayTones** [reply]
setting

```
      createdAt: { type: Date },
      updatedAt: { type: Date },
```

in the schema model did indeed store the dates as the expected ISO format, thank you.

**Reply by mobsense** [reply]
Don't include them in your model -- we add them automatically. You are overriding our definitions and setting them to String which is the cause of the breakage.

**Comment by mobsense** [comment, marked as answer]
Don't set them to strings in your schema. Either don't specify them or set them to Date ('date')
